src/world.rs

```diff
@@ -1294,9 +1354,27 @@ impl ArchetypeSet {
             index,
         }
     }
+
+    pub fn try_clone(
+        &mut self,
+        cloner: &crate::cloning::Cloner,
+    ) -> Result<Self, TypeUnknownToCloner> {
+        Ok(Self {
+            index: self.index.clone(),
+            archetypes: {
+                let mut vec = Vec::with_capacity(self.archetypes.capacity());
+                for archetype in self.archetypes.iter_mut() {
+                    let cloned = archetype.try_clone(cloner)?;
+                    vec.push(cloned)
+                }
+                vec
+            },
```

Comment:
Something like

```suggestion
            archetypes: self.archetypes
                .iter()
                .map(|archetype| archetype.try_clone(cloner))
                .collect::<Result<Vec<_>, _>>()?,
```

should also work AFAIU.

Reply:
Ah, I need iter_mut(), but yes, you're right - no need to manually set capacity & loop here. Fixed.